final_output_nb.ipynb

```diff
@@ -1831,10 +2059,38 @@
     "    tempdf2[\"num_clauses\"] = ''\n",
     "    tempdf2[\"num_words\"] = ''\n",
     "    tempdf2[\"clause_num\"] = tempdf2.index + 1\n",
-    "    dfs.append(pd.concat([tempdf1, tempdf2]))\n",
+    "    tempdf2[\"UID\"] = row['UID'] # tjm added\n",
+    "    tempdf2[\"score\"] = row['score'] # tjm added\n",
+    "    dfs.append(pd.concat([tempdf1, tempdf2], sort=True)) # tjm added sort=True\n",
+    "\n",
+    "pd.concat(dfs).to_csv(\"debug.csv\", index = False)\n",
```

Comment:
The subsequent lines rebuild the debug.csv contents again. It is better to persist df_dfs to debug.csv. Remove this line which reads `pd.concat(dfs).to_csv("debug.csv", index = False)
` and add a line after building df_dfs completely which reads `df_dfs.to_csv("debug.csv")`  for better debugging